src/cljc/rems/common/application_util.cljc

```diff
@@ -27,3 +29,25 @@
 
 (defn is-handler? [application user]
   (contains? (workflow-handlers application) user))
+
+(defn- str-to-int [s]
+  #?(:clj (Integer/parseInt s)
+     :cljs (js/parseInt s 10)))
+
+(defn- printToConsole [s]
+  #?(:clj (println s)
+     :cljs (js/console.log s)))
+
+(defn parse-sortable-external-id [external-id]
+  ;; the idea is to parse all numbers from the string, then add them up to form an external id - 
+  ;; comes from a theory that it is somehow tied to the date or the numbers are created in sequence
+  (cond
+    (if (seq (re-seq #"\d+" external-id)) "true" "false") (reduce + (map (fn [str] (str-to-int str)) (re-seq #"\d+" external-id)))
```

Comment:
Why sum the numbers? This leads to funny sort orders, for example:

- ABC/2000.1
- ABC/2003.1
- ABC/2001.5

It's of course hard to sort arbitrary identifiers in general, but perhaps sorting the number sequences lexicographically would be a better solution. Luckily clojure vectors work nicely for this:

```
(map #(vec (re-seq #"\d+" %)) ["ABC/2000.1" "ABC/2003.1" "ABC/2001.5" "ABC/2001.11"])
  ==> (["2000" "1"] ["2003" "1"] ["2001" "5"] ["2001" "11"])
(sort [[2000 1] [2003 1] [2001 5] [2001 11]])
  ==> ([2000 1] [2001 5] [2001 11] [2003 1])
```

In any case, a test that sorts various lists of identifiers would be useful both to document the behaviour and as a place to add regression cases.

Reply:
ok, Ill redo it :)

Reply:
Doesn't this also lose all the strings for comparison purposes? Also test what happens with 0-prefixed numbers i.e. 001 and stuff like 4.00.00. See `THL/2492/4.00.00/2020`

Reply:
0-prefixed numbers should work fine when parsed into ints.

Reply:
Yes but we need tests to cover them.

Reply:
"Doesn't this also lose all the strings for comparison purposes?" - sorry @Macroz do you mean my solution?

Reply:
Any solution with `re-seq` of just digits.